src/Native/configure.cmake

```diff
@@ -196,13 +199,53 @@ check_function_exists(
     kqueue
     HAVE_KQUEUE)
 
+check_cxx_source_compiles(
+     "
+     #include <sys/types.h>
+     #include <netdb.h>
+
+     int main()
+     {
+         const void* addr;
+         socklen_t len;
+         int type;
+         struct hostent* result;
+         char* buffer;
+         size_t buflen;
+         struct hostent** entry;
+         int* error;
+         gethostbyaddr_r(addr,  len, type, result, buffer, buflen, entry, error);
+         return 0;
+     }
+     "
+     HAVE_GETHOSTBYADDR_R)
+
+check_cxx_source_compiles(
+     "
+     #include <sys/types.h>
+     #include <netdb.h>
+
+     int main()
+     {
+         const char* hostname;
+         struct hostent* result;
+         char* buffer;
+         size_t buflen;
+         struct hostent** entry;
+         int* error;
+         gethostbyname_r(hostname, result, buffer, buflen, entry, error);
+         return 0;
+     }
+     "
+     HAVE_GETHOSTBYNAME_R)
+
 check_function_exists(
-    gethostbyname_r
-    HAVE_GETHOSTBYNAME_R)
+    getaddrinfo
+    HAVE_GETADDRINFO)
 
 check_function_exists(
-    gethostbyaddr_r
-    HAVE_GETHOSTBYADDR_R)
+    getnameinfo
+    HAVE_GETNAMEINFO)
```

Comment:
We only returned error from copy_hostent function, the fallback/shim functions still exist and are wrapped under these postprocessor conditions. Should we change those to `#ifdef __NETBSD__` and revert configure.cmake and pal_config.h.in?